loader/fixups.c

```diff
@@ -39,16 +39,48 @@ SYS_INIT(disable_bootloader_mpu, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAU
 SYS_INIT(disable_mpu_rasr_xn, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
 #endif
 
-#if defined(CONFIG_BOARD_ARDUINO_GIGA_R1) && defined(CONFIG_VIDEO)
+
+#if defined(CONFIG_BOARD_ARDUINO_GIGA_R1)  && defined(CONFIG_INPUT_GT911_INTERRUPT)
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+
+#include <zephyr/input/input.h>
+
+// experiment to try to capture touch screen events
+void (*_giga_touch_callback)(struct input_event *evt, void *user_data) = 0;
+
+void registerGigaTouchCallback(void (*cb)(struct input_event *evt, void *user_data))
+{
+	_giga_touch_callback = cb;
+}
+
+
+void touch_event_callback(struct input_event *evt, void *user_data)
+{
+    //printk("touch_event_callback(%p %p): %p %u %u %u %d\n", evt, user_data,
+    //        evt->dev, evt->sync, evt->type, evt->code, evt->value);
+	if (_giga_touch_callback)
+		(*_giga_touch_callback)(evt, user_data);
+}
+
+static const struct device *const touch_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_touch));
+INPUT_CALLBACK_DEFINE(touch_dev, touch_event_callback, NULL);
+
+#endif
+
+#if (defined(CONFIG_BOARD_ARDUINO_GIGA_R1) || defined(CONFIG_BOARD_ARDUINO_PORTENTA_H7)) \
+		&& defined(CONFIG_VIDEO)
 #include <zephyr/kernel.h>
 #include <zephyr/device.h>
-#include <zephyr/drivers/clock_control.h>
 #include <zephyr/logging/log.h>
+#include <zephyr/drivers/clock_control.h>
 
 int camera_ext_clock_enable(void)
 {
 	int ret;
 	uint32_t rate;
+
```

Comment:
Please revert any other unrelated changes.

Reply:
Note I still needs the #if code here as it was earlier under similar.  The only change left is
that it also supports Portenta.  I suppose we could simply do another PR to change this if...